meeting-notes/2018/201Y8-11-12--go-core-dev-team-weekly.md

```diff
@@ -0,0 +1,206 @@
+   - Review HAMT directory refactor PR (https://github.com/ipfs/go-unixfs/pull/42).
+   - Review trickle DAG refactor PR (https://github.com/ipfs/go-unixfs/pull/10).
+ - Next (not necessarily this week):
+   - With the DAG walker merged finish up the DAG reader refactor using it (https://github.com/ipfs/go-unixfs/pull/12).
+   - Write mid-level document on the `ipfs files` API (https://github.com/ipfs/go-ipfs/pull/5052)
+   - MFS interface (https://github.com/ipfs/go-mfs/issues/3).
+
+@magik6k (not attending)
+ - Done:
+   - Running datastore benchmarks right now
+   - (Almost) have consensus on go-ipfs-files refactor
+   - Did some reviews
+ - Blocked:
+   - No, but will need some reviews this week
+ - Next:
+   - Run all the benchmarks, on all the things
+   - Finish go-ipfs-files stuff
+   - Write a doc on datastore performance
+
+@eingenito
+  - Done:
+    - Worked on [2019 Planning Doc](https://docs.google.com/document/d/11zw8H8NjUzMM5CJ1y9lpvDxmVrJ6SAOxsRsiLPZxDaI/edit#heading=h.3o5e0g37o69l)
+    - Did [screencast of same](https://drive.google.com/open?id=1MHzaHujcYfOhQEOd0PseViTUkqm6Ap5D)
+    - Worked on bitswap with Hannah
+    - Investigated benchmarking - so many options
+  - Next
+    - OKR projections
+    - Benchmarking - just choose one
+    
+  
+@warpfork
+  - Done:
+    - Worked on [2019 Planning Doc](https://docs.google.com/document/d/11zw8H8NjUzMM5CJ1y9lpvDxmVrJ6SAOxsRsiLPZxDaI/edit)
+    - There have been some discussions on how representations of serial zeros and nulls and whatnot should work: https://github.com/polydawn/refmt/issues/42
+    - https://github.com/ipld/specs/pull/72 merged woo (thank you mikeal)
+    - go-ipld-prime has a MutableNode interface now and it has one impl; a fluent wrapper interface layer for convenience; some tests
+    - did some dives on merkleCI agent models across platforms (punted to victorb for review)
+  - Blocked: n/a
+  - Next:
+    - flesh out (way) more tests in go-ipld-prime, finish several todos for recursive mutables
+    - build a first draft of traversers and zippers in go-ipld-prime
+    
+@bigs
+  - Done:
+    - Local kubernetes cluster with daemon images!
+    - Code review
+    - SECIO spec first pass
+  - Blockers/callout:
+    - Would love a link to a good spec
+  - Next:
+    - Begin publishing daemon images
+    - Create basic integration test library for daemons
+    - SECIO spec complete
+    
+@raulk (not attending):
+  - Done:
+    - Compiled my field notes from devcon4 into a 18-page doc! Will distribute once I write a TL;DR for everybody else ;-)
+    - Kademlia DHT spec: https://github.com/libp2p/specs/pull/108
+    - Good discussion on connection manager and Multistream 2.0.
+    - Cruising speed on dialer v2! Integrating feedback and hardening.
+    - Almost done with peerstore simplification. Relaxing strict db-level TTL expiration helps a lot.
+  - Blocked: nothing.
+  - Next:
+    - Finish dialer v2.
```

Comment:
@raulk not trying to start a bikeshed, but I believe that this dialerV2 is the same as js-libp2p-switch https://github.com/libp2p/js-libp2p-switch/

Reply:
Hm, I don’t think so. Dialerv2 moves to a composable design where the dial behaviour can be customised by changing components in a pipeline. It’s WIP here:  https://github.com/libp2p/go-libp2p-swarm/pull/88